internal/pagination/pagination.go

```diff
@@ -154,7 +147,7 @@ func PaginateRequest[T any, PbT ResponseItem](
 	// something to populate LastItemId and LastItemUpdatedTime with.
 	if len(resp.Items) > 0 || refreshToken != nil {
 		newRefreshToken := &pbs.ListRefreshToken{
-			CreatedTime:         timestamppb.New(newRefreshTokenCreateTime),
+			CreatedTime:         timestamppb.New(newTokenCreateTime),
```

Comment:
This is where the logic happens for listing deleted IDs without overlapping timestamps. The first request uses the current time from the system for the timestamp. Subsequent requests uses the time from the refresh token, getting the new timestamp from the DB transaction.